cocos/spine/lib/instantiated.ts

```diff
@@ -94,18 +91,31 @@ function initAsm (): Promise<void> {
 
 export function waitForSpineWasmInstantiation (): Promise<void> {
     const errorReport = (msg: any) => { error(msg); };
-    if (WASM_SUPPORT_MODE === WebAssemblySupportMode.MAYBE_SUPPORT) {
-        if (sys.hasFeature(sys.Feature.WASM)) {
-            return initWasm(spineWasmUrl).catch(errorReport);
+    return ensureWasmModuleReady().then(() => Promise.all([
+        import('external:emscripten/spine/spine.asm.js'),
+        import('external:emscripten/spine/spine.js.mem'),
+        import('external:emscripten/spine/spine.wasm.js'),
+        import('external:emscripten/spine/spine.wasm'),
```

Comment:
Why need both wasm and asm.js?

Reply:
we need asm.js as fallback 

the build tool would cull the asm.js module if we don't need it

Reply:
What happen if asm.js is culled?

Reply:
`'external:emscripten/spine/spine.asm.js'` module would be like
```ts
export default function () {}
```
`'external:emscripten/spine/spine.js.mem'` would be like
```ts
export default '';
```

Reply:
this is the culling details, see the `cullingContent` field 
https://github.com/cocos/cocos-ccbuild/blob/c9347edc879e12050ae91b98870a401b272755f2/src/build-engine/engine-js/rollup-plugins/external-wasm-loader.ts#L70-L125

Reply:
Ok, it is better add some comment. Or it is confused.

Reply:
I will add the details to the wasm document, every engine developer that use wasm needs to read this document first

https://github.com/cocos/3d-tasks/blob/master/design-docs/scripting/engine-wasm-module.md#%E5%BB%BA%E8%AE%AE%E5%91%BD%E5%90%8D%E6%A0%BC%E5%BC%8F